docker/environment.yml

```diff
@@ -3,16 +3,21 @@ channels:
   - conda-forge
 dependencies:
   - python
-  - osmnx=1.5.* # for constructing network graph from OpenStreetMap using Overpass API
+  - osmnx=1.5.*
```

Comment:
`1.9.*` is the latest minor release as of now, if it's worth upgrading this dep

Reply:
We could give it a go --- it may have been I pinned to 1.5.* to avoid refactoring code due to other changes in dependencies (I tried to make software as new as possible, while re-writing code as least as possible).  As above, I'll try the update and see how that goes with other dependencies/refactoring later in the week.

Reply:
Reflecting on this, your time is best served holding off on this one for now. We're working on some deprecations in advance of OSMnx v2 later this year. If it will require some refactoring in our codebase here to upgrade anyway, then we should just hold off until those deprecations for the OSMnx v2 API are all in place in a couple of months. That is, if you have to do some refactoring anyway, then might as well wait to just do it all at once rather than piecemeal.

Reply:
Sounds good @gboeing; thanks for heads up.  I am sure there will be more patches requiring updates here by that stage in any case, so will be a good opportunity to review things at that point and see if we can better leverage OSMnx functionality and related packages and drop some other dependencies perhaps.